src/components/App.js

```diff
@@ -0,0 +1,194 @@
+        "Description 2": "Maecenas tincidunt lacus at velit. Vivamus vel nulla eget eros elementum pellentesque. Quisque porta volutpat erat. Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla.",
+        "Description 3": "Nam dui. Proin leo odio, porttitor id, consequat in, consequat ut, nulla.",
+        "Description 4": "In hac habitasse platea dictumst. Etiam faucibus cursus urna.",
+        "Description 5": "Aliquam quis turpis eget elit sodales scelerisque. Mauris sit amet eros.",
+        "Inventory Amount": 0, 
+        "Is Customer Prime Member?": false,
+        // currently Prime Member does nothing. All shows as Prime
+        "Price": 355,
+        "ProductKey": 1,
+        "Review Total": 1,
+        "Shipping Price": 0,
+        "Title": "Pannewitz's Sedge",
+        "Total 1*": 1,
+        "Total 2*": null,
+        "Total 3*": null,
+        "Total 4*": null,
+        "Total 5*": null,
+        "Total Review Stars": 1
+      }]
+    }
+    
+  }
+
+  componentWillMount(){
+    this.serverRequest = axios.get(`http://localhost:4000/products/75`).then((res)=> 
+      this.setState({
+        data: res.data
+      })
+    ).catch((err) => {
+      console.log(err)
+    })
+  }
+
+  
+  /**
+   *  Renders average star based on review total/total review stars and floors it.
+   */
+    
+  averageStars = () => {
+    function round(number, precision) {
+      var shift = function (number, precision, reverseShift) {
+        if (reverseShift) {
+          precision = -precision;
+        }  
+        var numArray = ("" + number).split("e");
+        return +(numArray[0] + "e" + (numArray[1] ? (+numArray[1] + precision) : precision));
+      };
+      return shift(Math.round(shift(number, precision, false)), precision, true);
+    }
+    
+    let rounded = round(this.state.data[0]["Total Review Stars"] / this.state.data[0]["Review Total"], 1).toString()
+    if(rounded.length === 1){
+      rounded += ".0"
+    }
+    return rounded
+  }
+
+  /**
+   *     renders starIcon based on elements.
+   *     Stretch goal:  Make it so that it is possible to switch between high res and low res depending on device/res
+   */
+  renderStarIcon = () => {
+    let rounded = parseInt(this.averageStars());
+    if(rounded < .5 && rounded > 0){
+      return "05"
+    }
+    if(rounded > .5 && rounded <= 1.2){
+      return "10"
+    }
+    if(rounded > 1.3 && rounded <= 1.7){
+      return "15"
+    }
+    if(rounded > 1.8 && rounded < 2.2){
+      return "20"
+    }
+    if(rounded > 2.2 && rounded <= 2.7){
+      return "25"
+    }
+    if(rounded > 2.7 && rounded <= 3.2){
+      return "30"
+    }
+    if(rounded > 3.2 && rounded <= 3.7){
+      return "35"
+    }
+    if(rounded > 3.7 && rounded <= 4.2){
+      return "40"
+    }
+    if(rounded > 4.2 && rounded <= 4.7){
+      return "45"
+    }
+    if(rounded > 4.7) {
+      return "50"
+    }
+    return "00"
+  }
```

Comment:
This doesn't seem very DRY to me. What about rounding to the nearest .5 with `(Math.round(rating * 2) / 2).toFixed(1)`

https://stackoverflow.com/questions/19390644/round-number-to-nearest-5-decimal

Reply:
I agree that there is a better way of doing this! Will refactor in a separate branch!